controllers/competition_supervisor/competition_supervisor.py

```diff
@@ -71,9 +72,10 @@ def check_required_libraries(path: Path) -> None:
     missing, incorrect = [], []
 
     for package in path.read_text().splitlines():
-        package = package.partition('#')[0].strip()
-        if not package:
+        package_match = re.match(r"[\w|-]+==[\d|.]+", package)
```

Comment:
I wonder if we'd be better off using the actual parser for this?

https://stackoverflow.com/questions/49689880/proper-way-to-parse-requirements-file-after-pip-upgrade-to-pip-10-x-x mentions that `pip._internal.parse_requirements` does the job, and from some local testing with our file it does (`pkg_resources.parse_requirements` doesn't like the `-f` option).

Reply:
I would strongly recommend against using pip's internals to do this: https://pip.pypa.io/en/stable/user_guide/#using-pip-from-your-program (apart from anything else, if competitors are going to be using this, it's likely to run against many different versions of pip, and so more likely that someone will run into a pip API change that breaks it)

Could you do something like remove lines that look like arguments to pip (e.g. lines that start with `-`) before passing them to `pkg_resources`?

Regardless: in character classes (`[...]`), pipes are interpreted literally, so currently this would match a line like `foo|bar==12|3`.